xbmc/platform/darwin/tvos/TVOSTopShelf.mm

```diff
@@ -46,119 +46,140 @@
     if (!storeUrl)
       return;
-
     storeUrl = [storeUrl URLByAppendingPathComponent:@"RA" isDirectory:YES];
-    const BOOL isJailbroken = [tvosShared isJailbroken];
-    CLog::Log(LOGDEBUG, "TopShelf: using shared path {} (jailbroken: {})", storeUrl.path.UTF8String,
-              isJailbroken ? "yes" : "no");
 
-    auto sharedDefaults = [[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]];
-    auto sharedDictJailbreak = isJailbroken ? [[NSMutableDictionary alloc] initWithCapacity:2 + 2]
-                                            : nil; // for jailbroken devices
+    const BOOL isSandboxed = [tvosShared isSandboxed];
+    CLog::Log(LOGDEBUG, "[TopShelf] (sadnboxed: {}) Use storeUrl: {} (sandboxed: {})",
+              isSandboxed ? "yes" : "no", storeUrl.path.UTF8String);
 
-    // store all old thumbs in array
+    // Store all old thumbs in array
     auto fileManager = NSFileManager.defaultManager;
     auto filePaths =
         [NSMutableSet setWithArray:[fileManager contentsOfDirectoryAtPath:storeUrl.path error:nil]];
     std::string raPath = storeUrl.path.UTF8String;
-    CVideoThumbLoader thumbLoader;
 
+    // Shared dicts  (if we are sandboxed we use sharedDefaults, else we use sharedJailbreak)
+    auto sharedDefaults =
+        isSandboxed ? [[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]] : nil;
+    auto sharedJailbreak = isSandboxed ? nil : [[NSMutableDictionary alloc] initWithCapacity:2 + 2];
+
+    // Function used to add category items in TopShelf shared dict
+    CVideoThumbLoader thumbLoader;
     auto fillSharedDicts =
-        [&](CFileItemList& videoItems, NSString* videosKey, NSString* videosTitleKey,
-            uint32_t titleStringCode,
+        [&](CFileItemList& items, NSString* categorKey, uint32_t categoryTitleCode,
             std::function<std::string(CFileItemPtr videoItem)> getThumbnailForItem,
             std::function<std::string(CFileItemPtr videoItem)> getTitleForItem) {
-          if (videoItems.Size() <= 0)
+          if (items.Size() <= 0)
           {
-            // cleanup if there is no RA
-            [sharedDefaults removeObjectForKey:videosKey];
-            [sharedDefaults removeObjectForKey:videosTitleKey];
+            // If there is no item in this category, remove this dict from sharedDefaults
+            [sharedDefaults removeObjectForKey:categorKey];
             return;
           }
 
-          const int topShelfSize = std::min(videoItems.Size(), MaxItems);
-          auto videosArray = [NSMutableArray arrayWithCapacity:topShelfSize];
-          for (int i = 0; i < topShelfSize; ++i)
+          // Create dict for this category
+          auto categoryDict = [NSMutableDictionary dictionaryWithCapacity:2];
+
+          // Save category title in dict
+          auto categoryTitle = @(g_localizeStrings.Get(categoryTitleCode).c_str());
+          [categoryDict setObject:categoryTitle forKey:@"categoryTitle"];
```

Comment:
1. make `@"categoryTitle"` a constant as it's used more than once
2. `categoryDict[@"categoryTitle"] = categoryTitle;`
3. you could also avoid the temporary variable `categoryTitle`